templates/python-selenium/src/main.py

```diff
@@ -13,77 +12,91 @@
-    out of it, it will not work. Asynchronous execution is required for communication with Apify platform,
-    and it also enhances performance in the field of web scraping significantly.
+    """Main entry point for the Apify Actor.
+
+    This coroutine is executed using `asyncio.run()`, so it must remain an asynchronous function for proper execution.
+    Asynchronous execution is required for communication with Apify platform, and it also enhances performance in
+    the field of web scraping significantly.
     """
     async with Actor:
-        # Read the Actor input
+        # Retrieve the Actor input, and use default values if not provided.
         actor_input = await Actor.get_input() or {}
         start_urls = actor_input.get('start_urls', [{'url': 'https://apify.com'}])
         max_depth = actor_input.get('max_depth', 1)
 
+        # Exit if no start URLs are provided.
         if not start_urls:
             Actor.log.info('No start URLs specified in actor input, exiting...')
             await Actor.exit()
 
-        # Enqueue the starting URLs in the default request queue
-        default_queue = await Actor.open_request_queue()
+        # Open the default request queue for handling URLs to be processed.
+        request_queue = await Actor.open_request_queue()
+
+        # Enqueue the start URLs with an initial crawl depth of 0.
         for start_url in start_urls:
             url = start_url.get('url')
             Actor.log.info(f'Enqueuing {url} ...')
-            await default_queue.add_request({'url': url, 'userData': {'depth': 0}})
+            request = Request.from_url(url, user_data={'depth': 0})
+            await request_queue.add_request(request)
 
-        # Launch a new Selenium Chrome WebDriver
+        # Launch a new Selenium Chrome WebDriver and configure it.
         Actor.log.info('Launching Chrome WebDriver...')
         chrome_options = ChromeOptions()
+
         if Actor.config.headless:
             chrome_options.add_argument('--headless')
+
         chrome_options.add_argument('--no-sandbox')
         chrome_options.add_argument('--disable-dev-shm-usage')
         driver = webdriver.Chrome(options=chrome_options)
 
+        # Test WebDriver setup by navigating to an example page.
         driver.get('http://www.example.com')
         assert driver.title == 'Example Domain'
 
-        # Process the requests in the queue one by one
-        while request := await default_queue.fetch_next_request():
-            url = request['url']
-            depth = request['userData']['depth']
+        # Process the URLs from the request queue.
+        while request := await request_queue.fetch_next_request():
+            url = request.url
+            depth = request.user_data['depth']
             Actor.log.info(f'Scraping {url} ...')
 
             try:
-                # Open the URL in the Selenium WebDriver
+                # Navigate to the URL using Selenium WebDriver.
```

Comment:
It probably won't help much with anything, but maybe we could wrap `driver.get` with `asyncio.to_thread` here?